compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -73,158 +73,71 @@ FailureOr<std::array<uint32_t, 3>> getPackedSize(linalg::LinalgOp linalgOp,
-  // There is no corresponding unpack for the specified pack operation
-  // 0 is used when unpack is empty
-  SmallVector<bool> unpackEmpty = {false};
-  SmallVector<SmallVector<int64_t>> innerPerm = {{1, 0}};
-  SmallVector<SmallVector<int64_t>> outerPerm = {{0, 1}};
-  auto packingConfigLevel1Attr = getPackingConfigPackingLevelAttr(
-      context, packedSizes, transposePackIndices, unpackEmpty, innerPerm,
-      outerPerm);
-
-  // Pack level => 2.
-  // packed size for [M, N, K, m, n, k]
-  auto maybePackedSize = getPackedSize(linalgOp, M, N, K);
-  if (failed(maybePackedSize)) return failure();
-  const auto [m0, n0, k0] = maybePackedSize.value();
-  packedSizes = {0, 0, 0, m0, n0, k0};
-
-  // Transpose A matrix from [M K m k m0 k0] to [M K k m m0 k0]
-  // Transpose B matrix from [K N k n n0 k0] to [K N n k k0 n0]
-  // Transpose C matrix from [M N m n m0 n0] to [M N n m m0 n0]
-  transposePackIndices = {0, 1, 2};
-  // Only the third pack operation has a corresponding unpack operation
-  unpackEmpty = {false, false, true};
-  innerPerm = {{0, 1}, {1, 0}, {0, 1}};
-  outerPerm = {{0, 1, 3, 2}, {0, 1, 3, 2}, {0, 1, 3, 2}};
-  auto packingConfigLevel2Attr = getPackingConfigPackingLevelAttr(
-      context, packedSizes, transposePackIndices, unpackEmpty, innerPerm,
-      outerPerm);
-
-  SmallVector<PackingConfigPackingLevelAttr> packingConfigLevelsVal = {
-      packingConfigLevel1Attr, packingConfigLevel2Attr};
-  auto packingConfigLevels =
-      PackingConfigPackingLevelsAttr::get(context, packingConfigLevelsVal);
-  auto config = PackingConfigAttr::get(context, packingConfigLevels);
-  setPackingConfig(linalgOp, config);
-
-  // ------------------------------------------------------
-  // -------------- Set lowering config -------------------
-  // ------------------------------------------------------
-  // Currently, assume working on a 2x2 AIE array, so the second level tile
-  // sizes should be (tileM0/2, tileN0/2). Considering the packing sizes, the
-  // adjusted tile sizes should be (tileM0/2/packedM1, tileN0/2/packedN1).
-  const auto tileM1 = findLargestFactor(tileM0 / m0, 16 * scaleFactor / m0);
-  const auto tileN1 = findLargestFactor(tileN0 / n0, 16 * scaleFactor / n0);
-  // Set tile size for K as constant 1, so that the packed outer K dimension
-  // is 1.
-  const int tileK = 1;
-
-  SmallVector<int64_t> TileSizeLevel0 = {tileM0, tileN0};
-  SmallVector<int64_t> TileSizeLevel1 = {0, 0, tileK};
-  SmallVector<int64_t> TileSizeLevel2 = {0, 0, 0, tileM1, tileN1, 0};
-  TileSizesListType tileSizes = {TileSizeLevel0, TileSizeLevel1,
-                                 TileSizeLevel2};
-  if (failed(setOpConfigAndEntryPointFnTranslation(
-          entryPointFn, linalgOp, tileSizes,
-          IREE::Codegen::DispatchLoweringPassPipeline::Custom))) {
-    return failure();
-  }
-  return success();
-}
-
-namespace {
 
 // Container class for the tiling at level 0 (the AIE shared memory) and level 1
```

Comment:
This comment should be updated if it's no longer just for the pad-pack approach. Would it be possible to say, in terms of  the 15 parameters, how many elements of A, B, and C will be resident in the AIE core and memtile memories for the different pipelines? 

I think for pad-pack for A it is:

memtile: M0 * K0 elements
AIE core memory : M1 * K1 elements. 

is that correct? i.e. the packing parameters don't enter the equations. Actually in my mind that was the definition of M0, M1, K0, K1...